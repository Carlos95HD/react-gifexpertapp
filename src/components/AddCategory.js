import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ( {setCategories} ) => {

  const [inputValue, setInputValue] = useState('')

  const handleInputChange = (e) => {
    // console.log( e.target.value )
    setInputValue(e.target.value)
  }

  const handleSubmit = (e) => {
    e.preventDefault()
    // console.log('Submit')
    if ( inputValue.trim().length > 2 ) {
      setCategories( catg => [inputValue,...catg] )
      setInputValue('')
    }
  }


  return (
    <form onSubmit={ handleSubmit }>
      <input 
      className='mb-4 mt-2 shadow-lg p-2 font-mono rounded focus:outline-none'
      placeholder='Search'
      type='text'
      value= { inputValue }
      onChange={ handleInputChange }
      />

    </form>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}