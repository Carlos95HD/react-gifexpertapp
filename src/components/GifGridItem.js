import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ( { title, url} ) => {

  return (
    <div className='card animate__animated animate__fadeIn'>
      <img src={url} alt={ title } className='text-center'/>
      <p className='font-sans text-gray-800'>{ title }</p>
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}