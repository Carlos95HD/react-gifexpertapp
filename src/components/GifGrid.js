import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import {GifGridItem }from './GifGridItem'
import PropTypes from 'prop-types'

export const GifGrid = ({ category }) => {

  const { data:images, loading} = useFetchGifs(category);

  // getGifs();

  return (
    <>
    <h3 className='animate__animated animate__fadeIn font-mono mb-5 mt-5 text-base text-2xl text-gray-800'>{category}</h3>

    {loading && <p className='font-mono animate__animated animate__flash text-gray-800'>Loading</p>}

    <div className='card-grid'>

      {
        images.map( img  => (
          <GifGridItem
          key={img.id}
          {...img}
          />
        ))
      }

    </div>
    </>

  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}