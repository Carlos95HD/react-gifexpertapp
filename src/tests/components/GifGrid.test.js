import React from 'react'
import { shallow } from 'enzyme'
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs'
jest.mock('../../hooks/useFetchGifs');

describe('Test de <GifGrid />', () => {

  const categoria = 'Libro'
  test('Snapshot de category', () => {

    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={ categoria } />);
    expect(wrapper ).toMatchSnapshot();

  });
  
  test('debe mostrar items cuando cargan imagen useFetchGifs', () => {

    const gifs = [{
      id:'ABC',
      url: 'https:/localhost/cualquier/cosa.jpg',
      title:'Cualquier cosa'
    },
    {
      id:'123',
      url: 'https:/localhost/cualquier/cosa.jpg',
      title:'Cualquier cosa'
    }]

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={ categoria } />)
     
    // expect( wrapper ).toMatchSnapshot()
    expect( wrapper.find('p').exists() ).toBe(false)
    expect( wrapper.find('GifGridItem').length ).toBe(gifs.length)

  });
  
})
