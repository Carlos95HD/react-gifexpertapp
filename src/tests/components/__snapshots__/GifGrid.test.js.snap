// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de <GifGrid /> Snapshot de category 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn font-mono mb-5 mt-5 text-base text-2xl text-gray-800"
  >
    Libro
  </h3>
  <p
    className="font-mono animate__animated animate__flash text-gray-800"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
